eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/GA4GHBeaconWSServerV2.java

```diff
@@ -218,6 +218,18 @@ public String checkErrorHelper(BeaconAlleleRequestBody request) {
             return "Either alternateBases ot variantType is required";
         }
 
+        if (request.getVariantType() != null) {
+            try {
+                VariantType variantType = VariantType.valueOf(request.getVariantType());
+            } catch (Exception e) {
+                String errorMessage = "Please provide a valid variant type from ";
+                for (VariantType vt : VariantType.values()) {
```

Comment:
return errorMessage + String.join(", ",  VariantType.values())

Reply:
This is giving an error actually

`Cannot resolve method 'join(java.lang.String, uk.ac.ebi.eva.commons.core.models.VariantType[])'`

Reply:
oh, ok, native arrays are not Iterable:
`return errorMessage + String.join(", ", Arrays.asList(VariantType.values()))`

Reply:
This was giving an error as well. I fixed it by using tostring function